src/Knp/Snappy/AbstractGenerator.php

```diff
@@ -516,7 +516,11 @@ protected function isAssociativeArray(array $array)
      */
     protected function executeCommand($command)
     {
-        $process = new Process($command, null, $this->env);
+        if (method_exists(Process::class, 'fromShellCommandline')) {
+            $process = Process::fromShellCommandline($command, null, $this->env);
```

Comment:
I think there's no reason to run the command in a shell: the command has no pipes, no conditionals, no shell variables whatsoever. The array format (for `$command`) is supported since Symfony v3.4. Older Symfony versions should be dropped IMO. Thus, we could instead remove the conditional here, and change the output of `buildCommand`.

Reply:
I used the `Process::fromShellCommandline` to not change the `AbstractGenerator::executeCommand` signature which would be a BC break.

Reply:
Ah right, `AbstractGenerator` isn't marked as an internal class.